### PR TITLE
Prune launcher dependencies

### PR DESCRIPTION
**Pull request by jvandort**
Prune some dependencies from the launcher so that it does not pull in unnecessary dependencies

* Delete `worker-services`. It had only one class and it wasn't even used by workers
* Remove some `:core` dependencies from `:launcher`
* Remove dependency on `:functional` from `:files`
* Remove dependency on `:functional` from `:logging`

Some more work is needed in this area, but this is a start

### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

**Comment by jvandort** [comment]
@bot-gradle test this

**Comment by jvandort** [comment]
Test failure is transitent

File: platforms/core-runtime/logging/src/main/java/org/gradle/internal/logging/events/PromptOutputEvent.java [hunk 1]
```diff
@@ -41,7 +42,25 @@ public void render(StyledTextOutput output) {
     /**
      * Converts the given text into the response object, or returns a new prompt to display to the user.
      */
-    public abstract Either<?, String> convert(String text);
+    public abstract PromptResult<?> convert(String text);
```
Comment (lptr) [inline platforms/core-runtime/logging/src/main/java/org/gradle/internal/logging/events/PromptOutputEvent.java:45]: ❓ What was wrong with `Either`?

Comment (jvandort) [reply platforms/core-runtime/logging/src/main/java/org/gradle/internal/logging/events/PromptOutputEvent.java:45]: It lives in `:functional`, which require Java 8

Comment (jvandort) [reply platforms/core-runtime/logging/src/main/java/org/gradle/internal/logging/events/PromptOutputEvent.java:45]: `Optional` is part of its interface so it cannot be easily moved out.

**Comment by lptr** [comment]
Can we get this merged?

**Comment by jvandort** [comment]
@lptr I was holding off for a green check but I'll consider your request one. I'll get this merged tomorrow

**Pull request merged**